okhttp/src/main/kotlin/okhttp3/OkHttpClient.kt

```diff
@@ -547,7 +586,7 @@ open class OkHttpClient internal constructor(
 
   class Builder() {
     internal var dispatcher: Dispatcher = Dispatcher()
-    internal var connectionPool: ConnectionPool = ConnectionPool()
+    internal var connectionPool: ConnectionPool? = null
```

Comment:
Hm, that's interesting. I'm not sure what to do about this, because the connection pool now needs a reference to the `routeDatabase` and several other fields from the Builder, whose final values aren't known until `build()` is called.

How important is this use case? Notably, `client.newBuilder()` was already going to cause a new connection pool to be created anyway, so this only affects cases where a single builder instance gets `build()` called on it multiple times.

Reply:
Jesse, you are talking about this case right?

```
val clientBuilder = OkHttpClient.Builder()

val client1 = clientBuilder.build()
val client2 = clientBuilder.build()
val client3 = client1.newBuilder().build()
```

I'd expect 1 & 3 to share a connection pool.

Reply:
Ok, I've fixed the code to cache the connection pool that it creates if one wasn't set by the caller. This should be good now.